app/jobs/course/material/download_job.rb

```diff
@@ -0,0 +1,30 @@
+class Course::Material::DownloadJob < ApplicationJob
+  include TrackableJob
+
+  protected
+
+  # Performs the download service.
+  #
+  # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+  #   downloaded.
+  # @param [String] filename The name of the zip file.
+  def perform_tracked(contents, filename)
+    zip_file = Course::Material::DownloadService.download_and_zip(contents)
+    redirect_to publish_file(zip_file, filename.gsub(/[:\/\\]/, ' ') + '.zip')
+  end
+
+  # Make the file public visible.
+  #
+  # @param [String] file The path to the local file.
+  # @param [String] name The name of public visible file.
+  # @return [String] The url to the public visible file
+  def publish_file(file, name)
+    # TODO: Support remote path and publish the file on the load balancing server.
```

Comment:
@lowjoel I added a publish_file method. It supposes to copy downloaded zip files to cache or load balancing server. 

Current version only works in local (i.e. proxy server is same with app server), I need to find a way to specify a remote host though.